grpcgcp/gcp_balancer.go

```diff
@@ -379,7 +382,20 @@ func (gb *gcpBalancer) getSubConnRoundRobin() *subConnRef {
 	if len(gb.scRefList) == 0 {
 		gb.newSubConn()
 	}
-	return gb.scRefList[atomic.AddUint32(&gb.rrRefId, 1)%uint32(len(gb.scRefList))]
+	scRef := gb.scRefList[atomic.AddUint32(&gb.rrRefId, 1)%uint32(len(gb.scRefList))]
+
+	// Wait until SubConn is ready.
+	for gb.scStates[scRef.subConn] != connectivity.Ready {
```

Comment:
This loop would never terminate in case subconn is not ready. What is the general idea for the case where subconn takes too long to connect?

Reply:
Good catch! Fixed in the new commit